src/Microsoft.ComponentDetection.Detectors/npm/NpmComponentDetector.cs

```diff
@@ -88,10 +89,38 @@ protected virtual bool ProcessIndividualPackageJTokens(string filePath, ISingleF
             return false;
         }
 
-        if (enginesToken != null && enginesToken["vscode"] != null)
+        if (enginesToken != null)
         {
-            this.Logger.LogInformation("{NpmPackageName} found at path {NpmPackageLocation} represents a built-in VS Code extension. This package will not be registered.", name, filePath);
-            return false;
+            var containsVsCodeEngine = false;
+            if (enginesToken.Type == JTokenType.Array)
+            {
+                var engineStrings = enginesToken
+                    .Children()
+                    .Where(t => t.Type == JTokenType.String)
+                    .Select(t => t.ToString())
+                    .ToArray();
+                if (engineStrings.Any(e => e.Contains("vscode")))
+                {
+                    containsVsCodeEngine = true;
+                }
+            }
+            else if (enginesToken.Type == JTokenType.Object)
+            {
+                if (enginesToken["vscode"] != null)
+                {
+                    containsVsCodeEngine = true;
+                }
+            }
+            else
+            {
+                throw new NotImplementedException($"Unsupported engines token type {enginesToken}");
+            }
```

Comment:
I'd move `containsVsCodeEngine` up in scope, out of this if/else, and avoid throwing this exception.

Reply:
Done.